engine/routes.py

```diff
@@ -321,28 +431,34 @@ def _delimited_fs(delimiter):
 
 
 @app.route('/aggregate', methods=['POST'])
-@set_timeout(timeout=app.config.get('AGGREGATION_TIMEOUT_SECONDS'))
-def aggregate_async():
+# timeout set on _aggregation_helper so that async job dir is accessible to decorator
+def aggregation_wrapper():
     if app.config['AGGREGATE']:
         input_data = request.get_json()
         async_job = input_data.get("async_job")
         request_id = input_data.get("requestUUID")
-
-        # Verify the supplied path is valid before proceeding
-        raise_invalid_path(app.config['FINAL_ASYNC_DIR'], async_job)
-
-        log.info("<%s> Performing aggregation on asynchronous job %s", request_id, async_job)
-        st = time.time()
-        util.aggregation.aggregate(async_job, request_id=request_id)
-        et = time.time() - st
-        log.info("<%s> Done performing aggregation on asynchronous job %s took %s seconds", request_id, async_job, et)
+        _aggregation_helper(async_job, request_id)
         message = "aggregation complete"
     else:
         message = "aggregation disabled"
 
     return jsonify({'code': 200, 'message': message})
 
 
+@set_inactivity_timeout(is_active=util.aggregation.is_aggregation_progressing,
+                        poll_period=app.config.get('AGGREGATION_ACTIVITY_POLL_PERIOD'),
+                        max_runtime=app.config.get('AGGREGATION_MAX_RUNTIME'))
+def _aggregation_helper(async_job_dir, request_id):
+    # Verify the supplied path is valid before proceeding
+    raise_invalid_path(app.config['FINAL_ASYNC_DIR'], async_job_dir)
```

Comment:
it may be good to be consistent with accessing the app.config either use config.get or config[...] and try not to mix.  i think config.get allows to set a fallback value in case the key is not defined so I usually preferr to use the .get method; i think the .get method also fails cleaner

Reply:
I think both usages are valid. I believe app.config.get('key') will return a default value if the key is not found in config. If no default is specified, it returns None. The app.config['key'] will raise a KeyError if the key is not found. Depending on the situation (i.e. whether or not a missing key can be tolerated by supplying a default), one or the other will make more sense. The two variants are each used in over a dozen files, so it would be a significant task to analyze the uses and try to converge on one.